### PR TITLE
Switch to GridTools version 1.1

### PR DESCRIPTION
**Pull request by havogt**
## Technical Description

Changes:
- requires c++14 for generated code.
- increased the hard-coded offset limit because k-caches are handled slightly different in GridTools internals.

With GT v1.1 C++ symbol names are longer which might break CUDA compilation for some stencils. This CUDA bug is fixed in CUDA 11.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
Most likely the failure in ptxas is the problem of too long symbol names. I don't know how to proceed here.

```
nvcc error   : 'ptxas' died due to signal 11 (Invalid memory reference)
nvcc error   : 'ptxas' core dumped
```

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
@jdahm Are you already compiling with GT v1.1 or would we potentially break your code?

File: gtclang/test/integration-test/CodeGen/CMakeLists.txt [hunk 1]
```diff
@@ -77,7 +77,7 @@ function(compile_target)
   endif()
 
   target_compile_definitions(${executable} PRIVATE -DOPTBACKEND=${backend})
-  target_compile_features(${executable} PRIVATE cxx_std_11)
+  target_compile_features(${executable} PRIVATE cxx_std_14)
```
Comment (jdahm) [inline gtclang/test/integration-test/CodeGen/CMakeLists.txt:80]: @havogt should this be inherited from the `GridTools::gridtools` target?

Comment (havogt) [reply gtclang/test/integration-test/CodeGen/CMakeLists.txt:80]: It should be inherited, I'll try to remove it.

**Comment by havogt** [comment]
New jenkins run is http://jenkins-mch.cscs.ch/view/PASCHA/job/dawn_gtclang_clang-gridtools_PR/634

**Comment by havogt** [comment]
Tests passed @jdahm. Ready to merge from my point of view.

**Comment by jdahm** [comment]
@havogt Passes on my VM! I'll merge right now along with the clang-gridtools changes.

**Pull request merged**